org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/XMLFoldingsTest.java

```diff
@@ -218,32 +218,206 @@ public void	testLimit()  {
 			/*19*/" </span>",
 			/*20*/"</div>",
 		};
-		assertRanges(input, new ExpectedIndentRange[] {r(0, 20), r(1, 19), r(2, 4), r(5, 12), r(6, 8), r(9, 11), r(13, 15), r(16, 18)}, "no limit", null);
-		assertRanges(input, new ExpectedIndentRange[] {r(0, 20), r(1, 19), r(2, 4), r(5, 12), r(6, 8), r(9, 11), r(13, 15), r(16, 18)}, "limit 8", 8);
-		assertRanges(input, new ExpectedIndentRange[] {r(0, 20), r(1, 19), r(2, 4), r(5, 12), r(6, 8), r(13, 15), r(16, 18)}, "limit 7", 7);
-		assertRanges(input, new ExpectedIndentRange[] {r(0, 20), r(1, 19), r(2, 4), r(5, 12), r(13, 15), r(16, 18)}, "limit 6", 6);
-		assertRanges(input, new ExpectedIndentRange[] {r(0, 20), r(1, 19), r(2, 4), r(5, 12), r(13, 15)}, "limit 5", 5);
-		assertRanges(input, new ExpectedIndentRange[] {r(0, 20), r(1, 19), r(2, 4), r(5, 12)}, "limit 4", 4);
-		assertRanges(input, new ExpectedIndentRange[] {r(0, 20), r(1, 19), r(2, 4)}, "limit 3", 3);
-		assertRanges(input, new ExpectedIndentRange[] {r(0, 20), r(1, 19)}, "limit 2", 2);
-		assertRanges(input, new ExpectedIndentRange[] {r(0, 20)}, "limit 1", 1);
+		XMLFoldingSettings settings = new XMLFoldingSettings ();
+        settings.setIncludeClosingTagInFold(false);
+		assertRanges(input, new ExpectedIndentRange[] {r(0, 19), r(1, 18), r(2, 3), r(5, 11), r(6, 7), r(9, 10), r(13, 14), r(16, 17)}, "no limit", null);
+		assertRanges(input, new ExpectedIndentRange[] {r(0, 19), r(1, 18), r(2, 3), r(5, 11), r(6, 7), r(9, 10), r(13, 14), r(16, 17)}, "limit 8", 8);
+		assertRanges(input, new ExpectedIndentRange[] {r(0, 19), r(1, 18), r(2, 3), r(5, 11), r(6, 7), r(13, 14), r(16, 17)}, "limit 7", 7);
+		assertRanges(input, new ExpectedIndentRange[] {r(0, 19), r(1, 18), r(2, 3), r(5, 11), r(13, 14), r(16, 17)}, "limit 6", 6);
+		assertRanges(input, new ExpectedIndentRange[] {r(0, 19), r(1, 18), r(2, 3), r(5, 11), r(13, 14)}, "limit 5", 5);
+		assertRanges(input, new ExpectedIndentRange[] {r(0, 19), r(1, 18), r(2, 3), r(5, 11)}, "limit 4", 4);
+		assertRanges(input, new ExpectedIndentRange[] {r(0, 19), r(1, 18), r(2, 3)}, "limit 3", 3);
+		assertRanges(input, new ExpectedIndentRange[] {r(0, 19), r(1, 18)}, "limit 2", 2);
+		assertRanges(input, new ExpectedIndentRange[] {r(0, 19)}, "limit 1", 1);
+	}
+
+	@Test
+	public void testFoldOneLevelWithincludeClosingTagInFoldAsTrue () {
+		String[] input = new String[] {
+			/*0*/"<html>",
+			/*1*/"Hello",
+			/*2*/"</html>"
+		};
+		XMLFoldingSettings settings = new XMLFoldingSettings ();
+        settings.setIncludeClosingTagInFold(true);
+		assertRanges(input,  new ExpectedIndentRange[] {r(0, 2)}, settings);
+	}
+
+	@Test
+	public void	testFoldTwoLevelWithincludeClosingTagInFoldAsTrue ()  {

```

Comment:
remove tabs with space (`public void test...`). Do the samething for other test methods